operator/helper/host_labeler.go

```diff
@@ -0,0 +1,125 @@
+package helper
+
+import (
+	"fmt"
+	"net"
+	"os"
+
+	"github.com/observiq/carbon/entry"
+	"github.com/observiq/carbon/errors"
+)
+
+// NewHostLabelerConfig returns a HostLabelerConfig with default values
+func NewHostLabelerConfig() HostLabelerConfig {
+	return HostLabelerConfig{
+		IncludeHostname: true,
+		IncludeIP:       true,
+		GetHostname:     getHostname,
+		GetIP:           getIP,
+	}
+}
+
+// HostLabelerConfig is the configuration of a host labeler
+type HostLabelerConfig struct {
+	IncludeHostname bool `json:"include_hostname,omitempty"     yaml:"include_hostname,omitempty"`
+	IncludeIP       bool `json:"include_ip,omitempty"     yaml:"include_ip,omitempty"`
+	GetHostname     func() (string, error)
```

Comment:
Is there a better place we can put `GetHostname` and `GetIP` overrides? Generally, I think config structs should represent the things actually in the config file.

Reply:
Additionally, test overrides probably shouldn't be part of the package's public interface

Reply:
I did this primarily for injection purposes during testing. Do you have another way you think we can do this? I wasn't a big fan of the variable override that we had before, and it also didn't allow us to test the IP labeling, which was omitted from the tests.

Reply:
They also started private, but I ended up having to make them public for the tests in host_metadata, since that was in a different package.

Reply:
They were also put on the config because the config builds the labeler, and as part of the build step, it looks up the IP and Hostname, since we don't want to do that on every call to Label. That's originally what the host_metadata operator was doing. In the build function on the config, it was calling those functions.

Reply:
Well, honestly, we're still not testing IP labeling. Really the _only_ thing this tests right now is if our "Include*" arguments work, but interesting logic to test is completely overridden right now. 

Right now, we're testing the same functionality in two places. It seems like we could get rid of the operator tests without losing any value. Then, we can make those private variables. Any reason you didn't like the overridable package-level variables rather than putting them on the struct? It feels really weird to have function fields on a config struct. Especially since I think that will fail when we try to marshal to YAML (can't represent a function as yaml).

Reply:
If I remove those tests from the operator, do we still get to keep the benchmark tests? I think the main reason I had for configurable functions is I didn't like the idea of a global variable that we swapped out for testing purposes and then had to remember to reset with a defer. It felt like a code smell to me. By making it something that could be configurable on the struct, you made it so you could swap in hostname and IP functions on a per config basis, which felt safer.